### ISSUE TITLE
Introduce ServiceCluster concept.

### ISSUE THREAD

**Issue by jpeach** [projectcontour/contour#2769]
**Please describe the problem you have**

The Contour DAG model needs a new concept to be able to represent the concept of an Envoy `ClusterLoadAssignment` that is backed by multiple (weighted) Kubernetes services.

The following is pretty rough, but I'm thinking of something like this to represent the load assignment:
```Go
type WeightedService struct {
       Weight      uint32
       ServiceName types.NamespacedName
       ServicePort v1.ServicePort
}

// ServiceCluster capture the set of Kubernetes Services that will
// compose the endpoints for a Envoy cluster.
type ServiceCluster struct {
       Services []WeightedService
}
```

Then we could rename the `dag.Service` to something more specific (maybe `RouteService`).

Next, the  Kubernetes Service field in `RouteService` would be replaced by a `ServiceCluster` with 1 element, and accessors and invariants added as appropriate.

**Comment by jpeach** [comment]
This blocks https://github.com/projectcontour/contour/issues/2713

**Comment by jpeach** [comment]
Fixed in #2805
